tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.py

```diff
@@ -81,25 +92,26 @@
 df = df.Define("goodlep_pt", "lep_pt[good_lep]")\
        .Define("goodlep_eta", "lep_eta[good_lep]")\
        .Define("goodlep_phi", "lep_phi[good_lep]")\
-       .Define("goodlep_E", "lep_E[good_lep]")
+       .Define("goodlep_E", "lep_E[good_lep]")\
+       .Define("goodlep_type", "lep_type[good_lep]")
 
 # Select leptons with high transverse momentum
 df = df.Filter("goodlep_pt[0] > 25000 && goodlep_pt[1] > 15000 && goodlep_pt[2] > 10000")
 
 # Reweighting of the samples is different for "data" and "MC". This is the function to add reweighting for MC samples
 ROOT.gInterpreter.Declare("""
-double weights(float scaleFactor_1, float scaleFactor_2, float scaleFactor_3, float scaleFactor_4, float mcWeight_, double xsecs_, double sumws_, double lumi_)
+double weights(float scaleFactor_1, float scaleFactor_2, float scaleFactor_3, float scaleFactor_4, float scale, float mcWeight_, double xsecs_, double sumws_, double lumi_)
```

Comment:
Better to avoid parameter/variable names with trailing underscores. In many code style conventions, trailing underscored denote private data members, which is used a lot in the community (see for example CMSSW).